lib/graphql/batch.rb

```diff
@@ -17,7 +17,16 @@ def self.batch(executor_class: GraphQL::Batch::Executor)
 
     def self.use(schema_defn, executor_class: GraphQL::Batch::Executor)
       schema = schema_defn.target
-      if GraphQL::VERSION >= "1.6.0"
+      if schema.respond_to?(:interpreter?) && schema.interpreter?
```

Comment:
I actually find comparing versions to be easier to maintain.  That way it is easy to remove dead code when we bump the minimum supported graphql gem version.  It also makes it easier to debug breaking changes like if `schema.respond_to?(:interpreter?)` stopped returning `true`, since it wouldn't fallback to code for older versions.

```suggestion
      if Gem::Version.new(GraphQL::VERSION) >= Gem::Version.new('1.9.0.pre3')
        require_relative "batch/mutation_field_extension"
```

Feature detection makes more sense when trying to support distinct implementations of something, but here we depend on a specific gem.

By moving the require for `MutationFieldExtension` into here, we can both avoid the extra version check as well as avoid loading that code when the interpreter isn't being used.

Reply:
Other than that, LGTM

Reply:
Someone could be using 1.9 though without the interpreter so this is a little different behaviour. However, I think it's safe to do anyway since field extensions work regardless of the interpreter. 👍 to the require in here as well